src/saltext/azurerm/utils/azurerm.py

```diff
@@ -100,13 +100,20 @@ def _determine_auth(**kwargs):
 
     try:
         credentials = DefaultAzureCredential(authority=authority, **kwargs)
-    except ClientAuthenticationError:
-        raise SaltInvocationError(  # pylint: disable=raise-missing-from
-            "Unable to determine credentials. "
-            "A subscription_id with username and password, "
-            "or client_id, secret, and tenant or a profile with the "
-            "required parameters populated"
-        )
+    except:  # pylint: disable=W0702
+        if "client_id" and "tenant" and "secret" in kwargs:
+            credentials = get_identity_credentials(**kwargs)
+        else:
+            kwargs.pop("client_id")
```

Comment:
**issue (blocking):** if we're modifying the input kwargs, we should log that we're doing so... even if it's in debug

**issue (blocking):** the preceding logic doesn't check if `client_id` exists in kwargs, so it's possible we throw a `KeyError` here.